server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/projection/cassandra/CassandraMailQueueModule.java

```diff
@@ -0,0 +1,65 @@
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.rabbitmq.projection.cassandra;
+
+import static com.datastax.driver.core.DataType.bigint;
+import static com.datastax.driver.core.DataType.blob;
+import static com.datastax.driver.core.DataType.list;
+import static com.datastax.driver.core.DataType.map;
+import static com.datastax.driver.core.DataType.smallint;
+import static com.datastax.driver.core.DataType.text;
+import static com.datastax.driver.core.DataType.timestamp;
+import static com.datastax.driver.core.schemabuilder.SchemaBuilder.frozen;
+
+import org.apache.james.backends.cassandra.components.CassandraModule;
+
+public interface CassandraMailQueueModule {
+
+    CassandraModule MODULE = CassandraModule
+        .table(EnqueuedMailsTable.TABLE_NAME)
+        .comment("store enqueued mail")
+        .options(options -> options)
+        .statement(statement -> statement
+            .addPartitionKey(EnqueuedMailsTable.QUEUE_NAME, text())
+            .addPartitionKey(EnqueuedMailsTable.TIME_RANGE_START, timestamp())
+            .addPartitionKey(EnqueuedMailsTable.BUCKET_ID, smallint())
+            .addPartitionKey(EnqueuedMailsTable.MAIL_KEY, text())
+            .addColumn(EnqueuedMailsTable.MESSAGE_SIZE, bigint())
+            .addColumn(EnqueuedMailsTable.STATE, text())
+            .addColumn(EnqueuedMailsTable.HEADER_BLOB_ID, text())
+            .addColumn(EnqueuedMailsTable.BODY_BLOB_ID, text())
+            .addColumn(EnqueuedMailsTable.ATTRIBUTES, map(text(), blob()))
+            .addColumn(EnqueuedMailsTable.ERROR_MESSAGE, text())
+            .addColumn(EnqueuedMailsTable.SENDER, text())
+            .addColumn(EnqueuedMailsTable.RECIPIENTS, list(text()))
+            .addColumn(EnqueuedMailsTable.REMOTE_HOST, text())
+            .addColumn(EnqueuedMailsTable.REMOTE_ADDR, text())
+            .addColumn(EnqueuedMailsTable.LAST_UPDATED, timestamp())
+            .addUDTMapColumn(EnqueuedMailsTable.PER_RECIPIENT_SPECIFIC_HEADERS, text(), frozen(EnqueuedMailsTable.HEADER_TYPE)))
+
+        .table(FirstEnqueuedMailTable.TABLE_NAME)
+        .comment("store the first enqueued mail, serve for iterating")
+        .options(options -> options)
+        .statement(statement -> statement
+            .addPartitionKey(FirstEnqueuedMailTable.QUEUE_NAME, text())
+            .addColumn(FirstEnqueuedMailTable.TIME_RANGE_START, timestamp()))
+
```

Comment:
You Miss a `DeletedMail` table:

 - QueueName + MailKey as Primary key

This table will be updated upon dequeue and needs to be checked for each mail while browsing